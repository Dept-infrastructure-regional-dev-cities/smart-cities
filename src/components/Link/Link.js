import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import style from './Link.scss';

const Link = (props) => {
  const rel = props.external ? 'noopener noreferrer' : null;
  const target = props.external ? '_blank' : null;

  return (
    <a
      className={classnames(style.link, props.className)}
      href={props.href}
      rel={rel}
      target={target}
    >
      {props.children || props.href}
    </a>
  );
};

Link.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  external: PropTypes.bool,
  href: PropTypes.string.isRequired,
};

export default Link;
