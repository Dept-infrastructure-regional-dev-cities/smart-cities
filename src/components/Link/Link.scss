@import '~style/variables';

.link {
  text-decoration: none;
  border-bottom: 1px solid $color-highlight;
  color: inherit;
  transition: $animation-duration;

  &:hover {
    background: $color-primary--500;
    border-bottom-color: transparent;
    color: $color-white;
    //text-decoration: none;
  }
}
