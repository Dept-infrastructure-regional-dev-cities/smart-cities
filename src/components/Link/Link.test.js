import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

const defaultProps = {
  href: 'https://www.example.com',
};

it('should match Snapshot', () => {
  const component = shallow(
    <Link {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render the children', () => {
  const component = shallow(
    <Link {...defaultProps} >
      Example dot com
    </Link>
  );

  expect(component.text()).toBe('Example dot com');
});

it('should render the href as the text if no children are supplied', () => {
  const component = shallow(
    <Link {...defaultProps} />
  );

  expect(component.text()).toBe('https://www.example.com');
});

describe('when the external flag is not set', () => {
  it('should set no target', () => {
    const component = shallow(
      <Link  {...defaultProps} />
    );

    expect(component.prop('target')).toBe(null)
  });

  it('should add rel noopener noreferrer', () => {
    const component = shallow(
      <Link  {...defaultProps} />
    );

    expect(component.prop('rel')).toBe(null)
  });
});

describe('when the external flag is true', () => {
  it('should add target="_blank', () => {
    const component = shallow(
      <Link  {...defaultProps} external />
    );

    expect(component.prop('target')).toBe('_blank')
  });

  it('should add rel noopener noreferrer', () => {
    const component = shallow(
      <Link  {...defaultProps} external />
    );

    expect(component.prop('rel')).toBe('noopener noreferrer')
  });
});
