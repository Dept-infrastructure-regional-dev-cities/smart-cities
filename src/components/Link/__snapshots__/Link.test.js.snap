// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<a className=\\"link\\" href=\\"https://www.example.com\\" rel={{...}} target={{...}}>
  https://www.example.com
</a>"
`;
